Frends.AzureDataLake.ListFiles/Frends.AzureDataLake.ListFiles/ListFiles.cs

```diff
@@ -0,0 +1,104 @@
+﻿namespace Frends.AzureDataLake.ListFiles;
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+using Azure.Identity;
+using Azure.Storage.Files.DataLake;
+using Frends.AzureDataLake.ListFiles.Definitions;
+
+/// <summary>
+/// Main class of the Task.
+/// </summary>
+public class AzureDataLake
+{
+    /// <summary>
+    /// List files and directories from Azure Data Lake container.
+    /// [Documentation](https://tasks.frends.com/tasks/frends-tasks/Frends.AzureDataLake.ListFiles)
+    /// </summary>
+    /// <param name="source">Source connection parameters.</param>
+    /// <param name="options">Options for the task</param>
+    /// <param name="cancellationToken">Token generated by Frends to stop this task</param>
+    /// <returns>Object { List&lt;FileData&gt; FileList [ { string Name, string URL } ] }</returns>
+    public static async Task<Result> ListFiles([PropertyTab] Source source, [PropertyTab] Options options, CancellationToken cancellationToken)
+    {
+        return new Result(await ListFileHandler(source, options, cancellationToken));
+    }
+
+    private static async Task<List<FileData>> ListFileHandler(Source source, Options options, CancellationToken cancellationToken)
+    {
+        var fileSystemClient = await GetDataLakeContainer(source, cancellationToken);
+        var fileListing = new List<FileData>();
+
+        try
+        {
+            if (options.ListingStructure == ListingStructure.Flat)
```

Comment:
Lets remove the options.ListingStructure altogether and simply replace it with a flag `bool Recursive`. This will simplify code and make it easier to understand for the user.